### PR TITLE
Elementwise Quantization Linear Ops, QLinearAdd and QLinearMul

### PR DESCRIPTION
**Pull request by zhanghuanrong**
Support quantization linear element wise math ops, specifically
implement QLinearAdd and QLinearMul. Support tests for quantization
linear element wise math ops, specifically implement test for
QLinearAdd and QLinearMul.

### CODE REVIEW

File: onnxruntime/contrib_ops/cpu/quantize_linear_element_wise.cc [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  QLinearBroadcastLoop(bc, output, input0scalar, input1scalar, general, A_scale, B_scale, C_scale, static_cast<int>(A_zero_point), static_cast<int>(B_zero_point), static_cast<int>(C_zero_point));
+  return Status::OK();
+}
+
+template <typename T>
+Status QLinearAdd<T>::Compute(OpKernelContext* context) const {
+  return QLinearBroadcastTwo<T>(
+      *context,
+      [](EigenVectorMap<T> output, T input0, ConstEigenVectorMap<T> input1, float A_scale, float B_scale, float C_scale, int A_zero_point, int B_zero_point, int C_zero_point) {
+        float a_value = A_scale * (static_cast<int>(input0) - A_zero_point);
```
Comment (yufenglee) [inline onnxruntime/contrib_ops/cpu/quantize_linear_element_wise.cc:45]: >float a_value = A_scale * (static_cast<int>(input0) - A_zero_point); [](start = 8, length = 68)

There is no need to compute this for each of element of b if a is scalar

File: onnxruntime/contrib_ops/cpu/quantize_linear_element_wise.cc [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        output = (((((input1.array().template cast<int>() - B_zero_point).template cast<float>() * B_scale) + a_value) / C_scale).round() - static_cast<float>(C_zero_point)).template cast<T>();
+      },
+      [](EigenVectorMap<T> output, ConstEigenVectorMap<T> input0, T input1, float A_scale, float B_scale, float C_scale, int A_zero_point, int B_zero_point, int C_zero_point) {
+        float b_value = B_scale * (static_cast<int>(input1) - B_zero_point);
```
Comment (yufenglee) [inline onnxruntime/contrib_ops/cpu/quantize_linear_element_wise.cc:49]: >float b_value = B_scale * (static_cast<int>(input1) - B_zero_point); [](start = 8, length = 68)

same for the case that b is scalar

File: onnxruntime/contrib_ops/cpu/quantize_linear_element_wise.cc [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        output = (((((input0.array().template cast<int>() - A_zero_point).template cast<float>() * A_scale) + b_value) / C_scale).round() - static_cast<float>(C_zero_point)).template cast<T>();
```
Comment (yufenglee) [inline onnxruntime/contrib_ops/cpu/quantize_linear_element_wise.cc:50]: >A_zero_point [](start = 60, length = 12)

Does eigen support compute type short (int16_t)?

Comment (yufenglee) [inline onnxruntime/contrib_ops/cpu/quantize_linear_element_wise.cc:50]: > static_cast<float> [](start = 139, length = 19)

round already cast the 1st operand to int. There is no need to cast to float here.

File: onnxruntime/contrib_ops/cpu/quantize_linear_element_wise.cc (hunk 1 above)
Comment (zhanghuanrong) [reply onnxruntime/contrib_ops/cpu/quantize_linear_element_wise.cc:45]: It is calculated only once here.

File: onnxruntime/contrib_ops/cpu/quantize_linear_element_wise.cc (hunk 3 above)
Comment (zhanghuanrong) [reply onnxruntime/contrib_ops/cpu/quantize_linear_element_wise.cc:50]: For regitster, it should be same speed as int32.

File: onnxruntime/contrib_ops/cpu/quantize_linear_element_wise.cc (hunk 2 above)
Comment (zhanghuanrong) [reply onnxruntime/contrib_ops/cpu/quantize_linear_element_wise.cc:49]: it is calculated only once.

File: onnxruntime/contrib_ops/cpu/quantize_linear_element_wise.cc (hunk 3 above)
Comment (zhanghuanrong) [reply onnxruntime/contrib_ops/cpu/quantize_linear_element_wise.cc:50]: Yes, could not cast here. Yet just for sure.

Comment (zhanghuanrong) [reply onnxruntime/contrib_ops/cpu/quantize_linear_element_wise.cc:50]: as SIMD finally will align to float. which is 32 bits.

File: onnxruntime/contrib_ops/cpu/quantize_linear_element_wise.cc (hunk 2 above)
Comment (yufenglee) [reply onnxruntime/contrib_ops/cpu/quantize_linear_element_wise.cc:49]: you have a loop in line 14.

File: onnxruntime/contrib_ops/cpu/quantize_linear_element_wise.cc (hunk 3 above)
Comment (yufenglee) [reply onnxruntime/contrib_ops/cpu/quantize_linear_element_wise.cc:50]: so the cast is zero cost or not?

**Comment by zhanghuanrong** [comment]
Using new PR. Close this one.

**Pull request closed without merging**